src/scripts.js

```diff
@@ -1,21 +1,76 @@
+    let currentImageIndex = 0;
 
-  // TODO: (Issue #1) Add an event listener for each image in the gallery to show it in the modal when clicked
+    // TODO: (Issue #1) Add an event listener for each image in the gallery to show it in the modal when clicked
+    images.forEach((image, index) => {
+        image.addEventListener("click", () => {
+            currentImageIndex = index;
+            updateModalImage();
+            buttonStateIndex();
+            modalImage.src = image.src;
+            modal.classList.remove("hidden");
+            modal.classList.add("active");
+        });
+    });
 
-  // TODO: (Issue #2) Add event listeners for the prevBtn and nextBtn for modal navigation functionality
-  // Clean Code Hint: Issue #1 and #2 both deal with updating the modal image based on currentImageIndex. Consider creating a function to handle this functionality.
+    updateModalImage();
+    buttonStateIndex();
 
-  // TODO: (Issue #3) Add an event listener for the closeBtn to close the modal
+    // TODO: (Issue #2) Add event listeners for the prevBtn and nextBtn for modal navigation functionality
+    // Clean Code Hint: Issue #1 and #2 both deal with updating the modal image based on currentImageIndex. Consider creating a function to handle this functionality.
 
-  // TODO: (Issue #3) Add an event listener to close the modal when clicking the modal background
+    function updateModalImage() {
+        modalImage.src = images[currentImageIndex].src;
+        prevBtn.disabled = currentImageIndex === 0;
+        nextBtn.disabled = currentImageIndex === images.length - 1;
+    }
+
+    nextBtn.addEventListener('click', () => {
+        currentImageIndex = (currentImageIndex + 1) % images.length;
+        updateModalImage()
+        buttonStateIndex()
+    });
+
+    prevBtn.addEventListener('click', () => {
+        currentImageIndex = (currentImageIndex - 1 + images.length) % images.length;
+        updateModalImage();
+        buttonStateIndex()
+    });
+
+    // TODO: (Issue #3) Add an event listener for the closeBtn to close the modal
+    closeBtn.addEventListener("click", () => {
+        modalImage.setAttribute("src", "");
+        modal.classList.remove("active");
+        modal.classList.add("hidden");
+    });
+
+    // TODO: (Issue #3) Add an event listener to close the modal when clicking the modal background
+    if (modal) {
+        modal.addEventListener('click', (event) => {
+            if (event.target === modal) {
+                modal.classList.remove('active');
+                modal.classList.add('hidden')
+            }
+        });
+    }
+
+    // TODO: (Issue #4) Create a function to update the state of the navigation buttons based on currentImageIndex
+    function buttonStateIndex() {
+
+        if (currentImageIndex === 0){
+            prevBtn.disabled = true;
+        }
+
+        if (currentImageIndex === images.length - 1){
+            nextBtn.disabled = true;
+        }
+    }
 
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏ you have two separate if conditions checking for the first and last images. You could combine these checks to make the code more concise, like this:

```js
function buttonStateIndex() {
  prevBtn.disabled = currentImageIndex === 0;
  nextBtn.disabled = currentImageIndex === images.length - 1;
}
```